x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1587,4 +1590,25 @@ private static LogicalPlan normalize(Aggregate aggregate, AttributeMap<Expressio
             return changed.get() ? new Aggregate(aggregate.source(), aggregate.child(), aggregate.groupings(), newAggs) : aggregate;
         }
     }
+
+    /**
+     * Fold the arms of {@code CASE} statements.
+     * <pre>{@code
+     * EVAL c=CASE(true, foo, bar)
+     * }</pre>
+     * becomes
+     * <pre>{@code
+     * EVAL c=foo
+     * }</pre>
+     */
+    static class PartiallyFoldCase extends OptimizerRules.OptimizerRule<Eval> {
```

Comment:
Case can appear outside Eval as well such as argument inside a function - so  simply make the rule for the expression itself, in this case Case (pun intended):
```
static class SimplifyCase extends OptimizerExpressionRule<Case> {

        protected Expression rule(Case c) {
               return  c.trimConditions();
        }
}
```

Reply:
++